pkg/azure/provider/helpers/driver.go

```diff
@@ -629,7 +652,21 @@ func createVMCreationParams(providerSpec api.AzureProviderSpec, imageRef armcomp
 		Zones:    getZonesFromProviderSpec(providerSpec),
 		Name:     &vmName,
 		Identity: getVMIdentity(providerSpec.Properties.IdentityID),
-	}, nil
+	}
+
+	if prop := providerSpec.Properties.SecurityProfile; prop != nil {
+		vm.Properties.SecurityProfile = &armcompute.SecurityProfile{
+			SecurityType: to.Ptr(armcompute.SecurityTypes(prop.SecurityType)),
+			UefiSettings: &armcompute.UefiSettings{
```

Comment:
Can you please elaborate the requirements which specifically only wishes to enable `TPM` and not `SecureBoot`? Can this also be made part of the API with a default value?

Reply:
`SecureBoot` requires that only trusted/signed components run during node boot. A counter-example of that would be for example Gardenlinux that would currently not be able to support that. Hence the TPM is an absolute necessity (learned the hard way that ConfidentialVMs need to have that enabled to start), but secure boot is "optional".

Reply:
For a lack of better place I will write this here. It related to both your comment here @unmarshall, as well as the previous at https://github.com/gardener/machine-controller-manager-provider-azure/pull/146#discussion_r1591791647.

> Can this also be made part of the API with a default value?

This PR introduces a little bit of implicit "smartness" by inferring options. Here, not all VM properties that are used are exposed to the public Spec, but are inferred from the `securityType`. Terraform does something similar, inferring the securityType from the OS's encryption type - [ref](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine). 

What I wanted to avoid for now is expose too many options without clear validation of what is allowed and what not. 
By exposing `SecureBoot` to the API, I would probably need to also expose UEFI settings. This begs the question of where should this "inference" reside. If I do expose these options, then I would move the inference to the provider-extension. Which is okay as long as we agree on it. 

The inference is necessary now because:
1. AFAIK, the UEFI settings only make sense if you use a `securityType` (either one type works).
2. The SecurityType infers the OS encryption (or the other around for TF).

I didn't have a clear answer to this question and I implemented the inference inside mcm-azure, but maybe it is not the correct place. WDYT ?

Reply:
I think it's a good idea not to expose too many options (like UEFI) to the user for now.
But to keep it flexible I'd rather put the "smartness" into the provider-extension than the mcm.
This means we would define the defaults in the extension and just pass them down avoiding any logic in the mcm.
In that case though we are faced with the old question how we define the validation.
I'd say in this case since we keep the defaults in the extension, we need no validation in the mcm.

Reply:
>But to keep it flexible I'd rather put the "smartness" into the provider-extension than the mcm.

I agree. MCM provider Spec is not really exposed for direct usage by the consumer. Therefore when defining MCM azure provide API, one should have UEFI as part of the API. In gardener context the correct values will be set by `gardener-extension-provider`. Please also remember that MCM is also used outside gardener and therefore inferencing it is not the best thing to do.

Reply:
Should be addressed with last commit. @hebelsan @unmarshall PTAL